ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -90,6 +101,21 @@ public PendingTransactions(
             "Count of transactions removed from the transaction pool",
             "source",
             "operation");
+
+    timerUtil.setPeriodic(transactionEvictionIntervalMs, this::evictOldTransactions);
+  }
+
+  private boolean applyEvictionThreshold(final TransactionInfo transaction) {
+    final long now = System.currentTimeMillis();
+    return now - transaction.getAddedToPoolAt().getEpochSecond() > transactionEvictionIntervalMs;
+  }
+
+  private void evictOldTransactions() {
+    synchronized (pendingTransactions) {
+      final List<TransactionInfo> transactionsToRemove =
+          prioritizedTransactions.stream().filter(this::applyEvictionThreshold).collect(toList());
```

Comment:
This is basically just style but I'd inline `applyEvictionThreshold` here and calculate the cut-off time just once. That makes the keep/remove condition really straight forward so it's a bit easier to follow.

Also, `System.currentTimeMillis` is a native method so incurs JNI overhead - we don't want to call that for every transaction in the pool if we can help it given we're inside a synchronized block.

Reply:
I'd still recommend this change.